hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/SparkUpsertCommitActionExecutor.java

```diff
@@ -44,6 +45,7 @@ public SparkUpsertCommitActionExecutor(HoodieSparkEngineContext context,
   @Override
   public HoodieWriteMetadata<JavaRDD<WriteStatus>> execute() {
     return SparkWriteHelper.newInstance().write(instantTime, inputRecordsRDD, context, table,
-        config.shouldCombineBeforeUpsert(), config.getUpsertShuffleParallelism(), this, true);
+            config.shouldCombineBeforeUpsert(), config.getUpsertShuffleParallelism(),
```

Comment:
when the delta streamer say is run with this code, it will pass the schema in and end up calling the preCombine(payload, schema) method. but it may not be present on the older payload implementations. This is tricky. but we need to find a way to not have this break.. any ideas?